src/PluggableDataType.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace SMW;
+
+/**
+ * @license GNU GPL v2+
+ * @since 3.1
+ *
+ * @author mwjames
+ */
+interface PluggableDataType {
+
+	/**
+	 * Returns the type_id and is by convention for an third-party extension
+	 * indicated by two underscores such as `__some_type`.
+	 *
+	 * @since 3.1
+	 *
+	 * @return string
+	 */
+	public function getTypeId();
+
+	/**
+	 * Returns the class name or callable associated with the type_id.
+	 *
+	 * @since 3.1
+	 *
+	 * @return string|callable
+	 */
+	public function getClass();
```

Comment:
I'm having some problems understanding what this is about, which suggests the naming is not super.